### DISCUSSION TITLE
A way to turn off autometrics instrumentation on the fly

### DISCUSSION THREAD

**Discussion by IvanMerrill** [autometrics-dev/autometrics-shared#80] [Ideas]
@pradeepcdlm raised the idea of being able to switch off the autometrics instrumentation on the fly. The suggestion was that it could be used to rule out autometrics in case of any performance issues. He'd suggested having some config somewhere that allows for this. Right now disabling autometrics would require code changes and a full deploy, which seems seems quite heavy. 

I wonder if this might be more useful as well for people when moving code through different environments. The ability to switch on metrics for dev testing, maybe turn them off on another environment but then enable them for production sounds like it might be useful.

**Comment by emschwartz** [comment]
This seems simple enough to add. Would something like an `AUTOMETRICS_DISABLE=true` environment variable work?

**Reply by IvanMerrill** [reply]
I think so, as long as this variable is checked semi-regularly so that you can just change this value and disable it all without needing to do anything else.

**Reply by emschwartz** [reply]
I think most systems only load the environment variables at startup. Given how most people run software today (whether it's on Kubernetes or a platform as a service), wouldn't you just change the values and restart the containers?

**Reply by emschwartz** [reply]
That would address the issue of being able to change it without making code changes or rebuilding anything, and also the point around different environments behaving differently.

**Reply by IvanMerrill** [reply]
This is a good point! Yes, I think you're right that this would be a suitable solution.
